config/routes.rb

```diff
@@ -17,9 +17,9 @@
 
 
   get "/merchants/:id/bulk_discounts/new", to: "bulk_discounts#new", as: "new_merchant_bulk_discount"
-  get "/merchants/:id/bulk_discounts/:id", to: "bulk_discounts#show"
+  get "/merchants/:merchant_id/bulk_discounts/:id", to: "bulk_discounts#show"
```

Comment:
These needed the merchant_id because overwriting in bulk_discounts was happening with two :id sections happening one after the other.